networking_module/prepare_data.R

```diff
@@ -166,15 +171,38 @@ if (!is.null(last.date)) {
     }
   )
 
-  if (nrow(dat) > 0) {
-
-    dates <- c(as.character(min(dat$requestindate)), as.character(max(dat$requestindate)), settings$xroad$instance)
-    saveRDS(dates, path.dates)
+  dates <- c(NA, NA, settings$xroad$instance)
+  dat2 <- data.frame()
 
+  dat <- odbcFetchRows(con, max=fetchsize)$data %>% as.data.frame(stringsAsFactors=FALSE)
+  while (nrow(dat) > 0) {
+    colnames(dat) <- c(
+      "requestindate", "clientmembercode", "clientsubsystemcode",
+      "servicemembercode", "servicesubsystemcode", "servicecode"
+    )
     dat[is.na(dat)] <- ''
+    min_date <- min(dat$requestindate)
+    max_date <- max(dat$requestindate)
+    dates[1] <- min(dates[1], min_date, na.rm = TRUE)
+    dates[2] <- max(dates[2], max_date, na.rm = TRUE)
+
+    dat <- dat %>%
+      count(clientmembercode, clientsubsystemcode, servicemembercode, servicesubsystemcode, servicecode)
+
+    dat2 <- rbind(dat2, dat)
+    dat2 <- dat2 %>%
+      group_by(clientmembercode, clientsubsystemcode, servicemembercode, servicesubsystemcode, servicecode) %>%
+      summarize_at(vars(n), sum) %>%
```

Comment:
What is  `n`  in this context?  I cannot find such data frame column named `n` ?

Reply:
Found it. This `n` is produced by `count()` in line 190